### PR TITLE
docs: add fuzzing improvement report.

### PR DESCRIPTION
**Pull request by DavidKorczynski**
Signed-off-by: davkor <david@adalogics.com>

For an explanation of how to fill out the fields, please see the relevant section
in [PULL_REQUESTS.md](https://github.com/envoyproxy/envoy/blob/main/PULL_REQUESTS.md)

Commit Message: Adds the fuzzing work recently carried out on the Envoy project.
Additional Description: CC @asraa @htuch 
Risk Level: Zero. Adds  PDF document.
Testing: None.
Docs Changes: None.
Release Notes: None.
Platform Specific Features: None.
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by htuch** [comment]
Can you create `docs/security` and put both this and the audit PDF there? Thanks.

**Comment by htuch** [comment]
CC @phlax for docs organization

**Comment by phlax** [comment]
one small nit - can we also update the README.md at root of the repo that currently points to `docs/SECURITY_AUDIT.pdf`

File: README.md [hunk 1]
```diff
@@ -80,7 +80,9 @@ Google calendar is here: https://goo.gl/PkDijT
 
 ### Security Audit
 
-A third party security audit was performed by Cure53, you can see the full report [here](docs/SECURITY_AUDIT.pdf).
+There has been several third party engagements focused on Envoy security:
+* In 2018, a Security audit was performed by Cure53, you can see the full report [here](docs/security/audity_cure53_2018.pdf)
+* In 2021, fuzzing infrastructure improvements were performed by Ada Logics, you can see the full report [here](docs/security/fuzzer_improvement_adalogics_2021.pdf).
```
Comment (phlax) [inline README.md:85]: this is a little confusingly worded - perhaps something like an "...an audit of fuzzing infrastructure was performed by Ada Logics with recommendations for improvements...

**Review by phlax — APPROVED** [review]
minor nit - otherwise lgtm - thanks

File: README.md (hunk 1 above)
Comment (DavidKorczynski) [reply README.md]: How do you like it now? I tried to make it a bit shorter as well by removing the "you can see the full report" and instead linking with "full report".

Comment (phlax) [reply README.md]: much better thanks!

**Review by phlax — APPROVED** [review]
lgtm - thanks again!

**Comment by DavidKorczynski** [comment]
please hold on to this one for a few minutes - I discovered a typo in the report.

**Comment by DavidKorczynski** [comment]
FIxed the typo now, thanks.

**Review by asraa — COMMENTED** [review]
Yay!! Seems like there's trailing whitespace in the README.md?
From ci/run_envoy_docker.sh 'ci/do_ci.sh format_pre'

**Comment by phlax** [comment]
@DavidKorczynski im surprised the repo doesnt enforce this but you probs should verify your commit signature - see https://docs.github.com/en/github/authenticating-to-github/managing-commit-signature-verification

**Review by phlax — APPROVED** [review]
lgtm

**Review by htuch — APPROVED** [review]
LGTM, thanks!

**Pull request merged**